src/main/kotlin/de/code_freak/codefreak/service/UserService.kt

```diff
@@ -0,0 +1,29 @@
+package de.code_freak.codefreak.service
+
+import de.code_freak.codefreak.entity.User
+import de.code_freak.codefreak.repository.UserRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class UserService : BaseService() {
+  @Autowired
+  private lateinit var userRepository: UserRepository
+
+  @Transactional
+  fun getOrCreateUser(username: String, patch: User.() -> Unit): User {
+    val user = try {
+      getUser(username)
+    } catch (e: EntityNotFoundException) {
+      userRepository.save(User(username))
+    }
+    user.patch()
```

Comment:
Patching will only work the when the user is initially created, otherwise the entity is detached. Not sure if this is desirable

Reply:
Do you want to implement a configuration option for this?